### PR TITLE
feat: add frontend-plugin-framework `LogoSlot`

### PR DESCRIPTION
**Pull request by brian-smith-tcril**
### Testing

I have created a testing PR to learner dash [here](https://github.com/openedx/frontend-app-learner-dashboard/pull/449) to get a sandbox deployed. If you visit https://app.pr-449-550b00.sandboxes.opencraft.hosting/learner-dashboard/ you should see the logo links to `https://openedx.org/`.

### Changelog
* Removed the `Logo` component, renamed `LinkedLogo` to `Logo`
  * I could not find any code paths in the org where the non-linked `Logo` component would be rendered.
* Removed the redundant `LinkedLogo` component from `LearningHeader.jsx`
* Added `className="logo"` directly to the `Logo` component
  * Every use of the `Logo` and `LinkedLogo` component included this. 
* Created a new `LogoSlot` component that wraps the `Logo` in a `frontend-plugin-framework` `PluginSlot`
  * Added a `README` with example `env.config.jsx` configs for replacing and modifying the logo
  * Replaced all uses of `Logo` with `LogoSlot`

### CODE REVIEW

File: src/plugin-slots/LogoSlot/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+The following `env.config.jsx` will replace the logo entirely (in this case with a centered 🗺️ `h1`)
+
+```jsx
+import { DIRECT_PLUGIN, PLUGIN_OPERATIONS } from '@openedx/frontend-plugin-framework';
+
+const config = {
+  pluginSlots: {
+    logo_slot: {
+      keepDefault: false,
+      plugins: [
```
Comment (davidjoy) [inline src/plugin-slots/LogoSlot/README.md:52]: Just as an aside while we're all talking about the syntax for plugins, I have to say... that's a whole lot of code to replace a logo. 😬

Comment (brian-smith-tcril) [reply src/plugin-slots/LogoSlot/README.md:52]: Yeah, I think people generally agree that FPF config feels a bit too verbose (hence the thread https://discuss.openedx.org/t/proposal-simplified-frontend-plugin-api-config/13312)

That being said, just changing the logo image itself is still very much possible via existing config methods - wrapping the Logo component in a PluginSlot doesn't change that

**Review by arbrandes — APPROVED** [review]
Can't find anything to object to!  Looks great!

**Comment by brian-smith-tcril** [comment]
@sarina do you have any ideas for what a good screenshot to showcase this plugin slot might be? The ability to choose which image to use for the logo has been around for a long time, so just showing a different logo doesn't feel like the best showcase.

I'm hoping to have an image that demonstrates that people can put *whatever* they want in there now. One idea I had was an svg logo that lets users set the color. Another idea would be a plugin that changes the logo image based on the date for holidays or seasons. If either of those sound good to you I'll gladly put something together and get some screenshots made. I'm also super open to other ideas!

**Pull request merged**